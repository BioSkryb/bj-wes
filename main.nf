nextflow.enable.dsl=2

include { printHeader; helpMessage } from './help' params ( params )
include { WES_WF } from './workflow/wes.nf' params ( params )

if ( params.help ) {
    helpMessage()
    exit 0
}


workflow {
    
    printHeader()
    
    if ( params.mode == "exome") {
        if ( params.exome_panel == "" ) {
            exit 1, "ERROR: exome_panel is not defined.\nPlease add --exome_panel. Available options are 'xGen Exome Hyb Panel v2' aka IDT, 'TruSight One', and 'TWIST'."
        }
    }
    
    // setting ch_reads from reads or input_csv
    if ( params.reads ) {
                
        ch_reads = Channel.fromFilePairs( params.reads , size: -1 , checkExists: true )
        
        ch_reads.ifEmpty{ exit 1, "ERROR: cannot find any fastq files matching the pattern: ${params.reads}\nMake sure that the input file exists!" }

    } else if ( params.input_csv ) {
        
        if ( params.platform.equalsIgnoreCase("Ultima") ) {
            ch_reads = Channel.fromPath( params.input_csv ).splitCsv( header:true )
                            .map { row -> [ row.biosampleName, [ row.cram, row.crai ] ] }

        } else {
                
            ch_reads = Channel.fromPath( params.input_csv ).splitCsv( header:true )
                        .map { row -> [ row.biosampleName, [ row.read1, row.read2 ] ] }
                    
        }
        
        ch_reads.ifEmpty{ exit 1, "ERROR: Input csv file is empty." }
    }
    
    
    ch_reads.view()
    ch_dummy_file = Channel.fromPath("$projectDir/assets/dummy_file.txt", checkIfExists: true).collect()
    // ch_dummy_file2 = Channel.fromPath("$projectDir/assets/dummy_file2.txt", checkIfExists: true).collect()

    
    WES_WF( 
                params.input_csv,
                ch_reads,
                ch_dummy_file,
                params.min_reads
             )
    
    
}

// OnComplete
workflow.onComplete{
    println( "\nPipeline completed successfully.\n\n" )
}

// OnError
workflow.onError{
    println( "\nPipeline failed.\n\n" )
}
