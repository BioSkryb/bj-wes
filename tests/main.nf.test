nextflow_pipeline {

    name "Test Workflow main.nf"
    script "main.nf"

    test("WES_test") {

        when {
            params {
                // define parameters here. Example: 
                publish_dir = "${outputDir}/results"
                timestamp = "test"
                skip_ado = true
                architecture = "x86"
                input_csv = "$baseDir/tests/data/inputs/input.csv"
            }
        }

        then {
            assertAll(
                // Check if the workflow was successful
                { assert workflow.success },

                // Verify existence of the multiqc report HTML file
                { assert new File("${outputDir}/results_test/multiqc/multiqc_report.html").exists()},

                // Check for a match in the all metrics MQC text file
                {assert snapshot (path("${outputDir}/results_test/secondary_analyses/metrics/nf-wes-pipeline_all_metrics_mqc.txt")).match("all_metrics_mqc")},

                // Check for a match in the selected metrics MQC text file
                {assert snapshot (path("${outputDir}/results_test/secondary_analyses/metrics/nf-wes-pipeline_selected_metrics_mqc.txt")).match("selected_metrics_mqc")},

                // Check for a match in the dedup_sentieonmetrics text file
                //{assert snapshot (path("${outputDir}/results_test/secondary_analyses/alignment/chr22_testsample1.dedup_sentieonmetrics.txt")).match("sentieonmetrics")},

                // Verify existence of the dnascope vcf file
                {assert new File("${outputDir}/results_test/secondary_analyses/variant_calls_dnascope/chr22_testsample1_dnascope.vcf.gz").exists()},

                // Verify existence of the snpEff vcf file
                {assert new File("${outputDir}/results_test/tertiary_analyses/variant_annotation/snpeff/chr22_testsample1_snpEff.ann.vcf").exists()},

                // Verify existence of the snpEff_version yml file
                {assert new File("${outputDir}/results_test/tertiary_analyses/variant_annotation/snpeff/snpEff_version.yml").exists()}

            )
        }

    }

}
