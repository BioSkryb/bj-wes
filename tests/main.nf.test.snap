{
    "selected_metrics_mqc": {
        "content": [
            "nf-wes-pipeline_selected_metrics_mqc.txt:md5,eacda23ee1015f71b69af430fb80ea2a"
        ],
        "timestamp": "2024-12-19T17:19:07+0000"
    },
    "all_metrics_mqc": {
        "content": [
            "nf-wes-pipeline_all_metrics_mqc.txt:md5,1dbdf86234064d849cfc1cb1f18a7f16"
        ],
        "timestamp": "2024-12-19T17:19:07+0000"
    }
}